### ISSUE TITLE
Build Fail on Nightly 2016-01-01

### ISSUE THREAD

**Issue by KodrAus** [erickt/rust-quasi#27]
I'm getting a number of compile errors on `nightly 2016-01-1`

```
<std macros>:3:1: 3:28 error: the trait `core::marker::Sync` is not implemented for the type `core::cell::UnsafeCell<Box<syntax::errors::emitter::Emitter + 'static>>` [E0277]
<std macros>:3 $ crate:: rt:: begin_unwind (
               ^~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/ashley/.cargo/registry/src/github.com-88ac128001ac3a9a/quasi-0.3.11/src/lib.rs:336:25: 336:36 note: in this expansion of panic! (defined in <std macros>)
/home/ashley/.cargo/registry/src/github.com-88ac128001ac3a9a/quasi-0.3.11/src/lib.rs:342:9: 342:39 note: in this expansion of panictry! (defined in /home/ashley/.cargo/registry/src/github.com-88ac128001ac3a9a/quasi-0.3.11/src/lib.rs)
<std macros>:3:1: 3:28 help: run `rustc --explain E0277` to see a detailed explanation
<std macros>:3:1: 3:28 note: `core::cell::UnsafeCell<Box<syntax::errors::emitter::Emitter + 'static>>` cannot be shared between threads safely
<std macros>:3:1: 3:28 note: required because it appears within the type `core::cell::RefCell<Box<syntax::errors::emitter::Emitter + 'static>>`
<std macros>:3:1: 3:28 note: required because it appears within the type `syntax::errors::DiagnosticBuilder<'_>`
<std macros>:3:1: 3:28 note: required by `std::sys_common::unwind::begin_unwind`
<std macros>:3:1: 3:28 error: the trait `core::marker::Sync` is not implemented for the type `core::cell::UnsafeCell<usize>` [E0277]
<std macros>:3 $ crate:: rt:: begin_unwind (
               ^~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/ashley/.cargo/registry/src/github.com-88ac128001ac3a9a/quasi-0.3.11/src/lib.rs:336:25: 336:36 note: in this expansion of panic! (defined in <std macros>)
/home/ashley/.cargo/registry/src/github.com-88ac128001ac3a9a/quasi-0.3.11/src/lib.rs:342:9: 342:39 note: in this expansion of panictry! (defined in /home/ashley/.cargo/registry/src/github.com-88ac128001ac3a9a/quasi-0.3.11/src/lib.rs)
<std macros>:3:1: 3:28 help: run `rustc --explain E0277` to see a detailed explanation
<std macros>:3:1: 3:28 note: `core::cell::UnsafeCell<usize>` cannot be shared between threads safely
<std macros>:3:1: 3:28 note: required because it appears within the type `core::cell::Cell<usize>`
<std macros>:3:1: 3:28 note: required because it appears within the type `core::cell::RefCell<Box<syntax::errors::emitter::Emitter + 'static>>`
<std macros>:3:1: 3:28 note: required because it appears within the type `syntax::errors::DiagnosticBuilder<'_>`
<std macros>:3:1: 3:28 note: required by `std::sys_common::unwind::begin_unwind`
/home/ashley/.cargo/registry/src/github.com-88ac128001ac3a9a/quasi-0.3.11/src/lib.rs:346:26: 346:38 error: cannot infer an appropriate lifetime for lifetime parameter `'a` due to conflicting requirements [E0495]
/home/ashley/.cargo/registry/src/github.com-88ac128001ac3a9a/quasi-0.3.11/src/lib.rs:346         panictry!(parser.parse_item())
                                                                                                                  ^~~~~~~~~~~~
/home/ashley/.cargo/registry/src/github.com-88ac128001ac3a9a/quasi-0.3.11/src/lib.rs:346:9: 346:39 note: in this expansion of panictry! (defined in /home/ashley/.cargo/registry/src/github.com-88ac128001ac3a9a/quasi-0.3.11/src/lib.rs)
/home/ashley/.cargo/registry/src/github.com-88ac128001ac3a9a/quasi-0.3.11/src/lib.rs:345:70: 347:2 note: first, the lifetime cannot outlive the destruction scope surrounding block at 345:69...
/home/ashley/.cargo/registry/src/github.com-88ac128001ac3a9a/quasi-0.3.11/src/lib.rs:345 pub fn parse_item_panic(parser: &mut Parser) -> Option<P<ast::Item>> {
/home/ashley/.cargo/registry/src/github.com-88ac128001ac3a9a/quasi-0.3.11/src/lib.rs:346         panictry!(parser.parse_item())
/home/ashley/.cargo/registry/src/github.com-88ac128001ac3a9a/quasi-0.3.11/src/lib.rs:347 }
/home/ashley/.cargo/registry/src/github.com-88ac128001ac3a9a/quasi-0.3.11/src/lib.rs:346:19: 346:38 note: ...so that references are valid when the destructor runs
/home/ashley/.cargo/registry/src/github.com-88ac128001ac3a9a/quasi-0.3.11/src/lib.rs:346         panictry!(parser.parse_item())
                                                                                                           ^~~~~~~~~~~~~~~~~~~
/home/ashley/.cargo/registry/src/github.com-88ac128001ac3a9a/quasi-0.3.11/src/lib.rs:346:9: 346:39 note: in this expansion of panictry! (defined in /home/ashley/.cargo/registry/src/github.com-88ac128001ac3a9a/quasi-0.3.11/src/lib.rs)
/home/ashley/.cargo/registry/src/github.com-88ac128001ac3a9a/quasi-0.3.11/src/lib.rs:345:70: 347:2 note: but, the lifetime must be valid for the anonymous lifetime #2 defined on the block at 345:69...
/home/ashley/.cargo/registry/src/github.com-88ac128001ac3a9a/quasi-0.3.11/src/lib.rs:345 pub fn parse_item_panic(parser: &mut Parser) -> Option<P<ast::Item>> {
/home/ashley/.cargo/registry/src/github.com-88ac128001ac3a9a/quasi-0.3.11/src/lib.rs:346         panictry!(parser.parse_item())
/home/ashley/.cargo/registry/src/github.com-88ac128001ac3a9a/quasi-0.3.11/src/lib.rs:347 }
/home/ashley/.cargo/registry/src/github.com-88ac128001ac3a9a/quasi-0.3.11/src/lib.rs:346:26: 346:38 note: ...so that types are compatible (expected `&mut syntax::parse::parser::Parser<'_>`, found `&mut syntax::parse::parser::Parser<'_>`)
/home/ashley/.cargo/registry/src/github.com-88ac128001ac3a9a/quasi-0.3.11/src/lib.rs:346         panictry!(parser.parse_item())
                                                                                                                  ^~~~~~~~~~~~
/home/ashley/.cargo/registry/src/github.com-88ac128001ac3a9a/quasi-0.3.11/src/lib.rs:346:9: 346:39 note: in this expansion of panictry! (defined in /home/ashley/.cargo/registry/src/github.com-88ac128001ac3a9a/quasi-0.3.11/src/lib.rs)
/home/ashley/.cargo/registry/src/github.com-88ac128001ac3a9a/quasi-0.3.11/src/lib.rs:350:26: 350:37 error: cannot infer an appropriate lifetime for lifetime parameter `'a` due to conflicting requirements [E0495]
/home/ashley/.cargo/registry/src/github.com-88ac128001ac3a9a/quasi-0.3.11/src/lib.rs:350         panictry!(parser.parse_pat())
                                                                                                                  ^~~~~~~~~~~
/home/ashley/.cargo/registry/src/github.com-88ac128001ac3a9a/quasi-0.3.11/src/lib.rs:350:9: 350:38 note: in this expansion of panictry! (defined in /home/ashley/.cargo/registry/src/github.com-88ac128001ac3a9a/quasi-0.3.11/src/lib.rs)
/home/ashley/.cargo/registry/src/github.com-88ac128001ac3a9a/quasi-0.3.11/src/lib.rs:349:60: 351:2 note: first, the lifetime cannot outlive the destruction scope surrounding block at 349:59...
/home/ashley/.cargo/registry/src/github.com-88ac128001ac3a9a/quasi-0.3.11/src/lib.rs:349 pub fn parse_pat_panic(parser: &mut Parser) -> P<ast::Pat> {
/home/ashley/.cargo/registry/src/github.com-88ac128001ac3a9a/quasi-0.3.11/src/lib.rs:350         panictry!(parser.parse_pat())
/home/ashley/.cargo/registry/src/github.com-88ac128001ac3a9a/quasi-0.3.11/src/lib.rs:351 }
/home/ashley/.cargo/registry/src/github.com-88ac128001ac3a9a/quasi-0.3.11/src/lib.rs:350:19: 350:37 note: ...so that references are valid when the destructor runs
/home/ashley/.cargo/registry/src/github.com-88ac128001ac3a9a/quasi-0.3.11/src/lib.rs:350         panictry!(parser.parse_pat())
                                                                                                           ^~~~~~~~~~~~~~~~~~
/home/ashley/.cargo/registry/src/github.com-88ac128001ac3a9a/quasi-0.3.11/src/lib.rs:350:9: 350:38 note: in this expansion of panictry! (defined in /home/ashley/.cargo/registry/src/github.com-88ac128001ac3a9a/quasi-0.3.11/src/lib.rs)
/home/ashley/.cargo/registry/src/github.com-88ac128001ac3a9a/quasi-0.3.11/src/lib.rs:349:60: 351:2 note: but, the lifetime must be valid for the anonymous lifetime #2 defined on the block at 349:59...
/home/ashley/.cargo/registry/src/github.com-88ac128001ac3a9a/quasi-0.3.11/src/lib.rs:349 pub fn parse_pat_panic(parser: &mut Parser) -> P<ast::Pat> {
/home/ashley/.cargo/registry/src/github.com-88ac128001ac3a9a/quasi-0.3.11/src/lib.rs:350         panictry!(parser.parse_pat())
/home/ashley/.cargo/registry/src/github.com-88ac128001ac3a9a/quasi-0.3.11/src/lib.rs:351 }
/home/ashley/.cargo/registry/src/github.com-88ac128001ac3a9a/quasi-0.3.11/src/lib.rs:350:26: 350:37 note: ...so that types are compatible (expected `&mut syntax::parse::parser::Parser<'_>`, found `&mut syntax::parse::parser::Parser<'_>`)
/home/ashley/.cargo/registry/src/github.com-88ac128001ac3a9a/quasi-0.3.11/src/lib.rs:350         panictry!(parser.parse_pat())
                                                                                                                  ^~~~~~~~~~~
/home/ashley/.cargo/registry/src/github.com-88ac128001ac3a9a/quasi-0.3.11/src/lib.rs:350:9: 350:38 note: in this expansion of panictry! (defined in /home/ashley/.cargo/registry/src/github.com-88ac128001ac3a9a/quasi-0.3.11/src/lib.rs)
/home/ashley/.cargo/registry/src/github.com-88ac128001ac3a9a/quasi-0.3.11/src/lib.rs:354:26: 354:37 error: cannot infer an appropriate lifetime for lifetime parameter `'a` due to conflicting requirements [E0495]
/home/ashley/.cargo/registry/src/github.com-88ac128001ac3a9a/quasi-0.3.11/src/lib.rs:354         panictry!(parser.parse_arm())
                                                                                                                  ^~~~~~~~~~~
/home/ashley/.cargo/registry/src/github.com-88ac128001ac3a9a/quasi-0.3.11/src/lib.rs:354:9: 354:38 note: in this expansion of panictry! (defined in /home/ashley/.cargo/registry/src/github.com-88ac128001ac3a9a/quasi-0.3.11/src/lib.rs)
/home/ashley/.cargo/registry/src/github.com-88ac128001ac3a9a/quasi-0.3.11/src/lib.rs:354:19: 354:37 note: first, the lifetime cannot outlive the method call at 354:18...
/home/ashley/.cargo/registry/src/github.com-88ac128001ac3a9a/quasi-0.3.11/src/lib.rs:354         panictry!(parser.parse_arm())
                                                                                                           ^~~~~~~~~~~~~~~~~~
/home/ashley/.cargo/registry/src/github.com-88ac128001ac3a9a/quasi-0.3.11/src/lib.rs:354:9: 354:38 note: in this expansion of panictry! (defined in /home/ashley/.cargo/registry/src/github.com-88ac128001ac3a9a/quasi-0.3.11/src/lib.rs)
/home/ashley/.cargo/registry/src/github.com-88ac128001ac3a9a/quasi-0.3.11/src/lib.rs:354:19: 354:37 note: ...so type `fn(&mut syntax::parse::parser::Parser<'_>) -> core::result::Result<syntax::ast::Arm, syntax::errors::DiagnosticBuilder<'_>>` of expression is valid during the expression
/home/ashley/.cargo/registry/src/github.com-88ac128001ac3a9a/quasi-0.3.11/src/lib.rs:354         panictry!(parser.parse_arm())
                                                                                                           ^~~~~~~~~~~~~~~~~~
/home/ashley/.cargo/registry/src/github.com-88ac128001ac3a9a/quasi-0.3.11/src/lib.rs:354:9: 354:38 note: in this expansion of panictry! (defined in /home/ashley/.cargo/registry/src/github.com-88ac128001ac3a9a/quasi-0.3.11/src/lib.rs)
/home/ashley/.cargo/registry/src/github.com-88ac128001ac3a9a/quasi-0.3.11/src/lib.rs:353:57: 355:2 note: but, the lifetime must be valid for the anonymous lifetime #2 defined on the block at 353:56...
/home/ashley/.cargo/registry/src/github.com-88ac128001ac3a9a/quasi-0.3.11/src/lib.rs:353 pub fn parse_arm_panic(parser: &mut Parser) -> ast::Arm {
/home/ashley/.cargo/registry/src/github.com-88ac128001ac3a9a/quasi-0.3.11/src/lib.rs:354         panictry!(parser.parse_arm())
/home/ashley/.cargo/registry/src/github.com-88ac128001ac3a9a/quasi-0.3.11/src/lib.rs:355 }
/home/ashley/.cargo/registry/src/github.com-88ac128001ac3a9a/quasi-0.3.11/src/lib.rs:354:26: 354:37 note: ...so that types are compatible (expected `&mut syntax::parse::parser::Parser<'_>`, found `&mut syntax::parse::parser::Parser<'_>`)
/home/ashley/.cargo/registry/src/github.com-88ac128001ac3a9a/quasi-0.3.11/src/lib.rs:354         panictry!(parser.parse_arm())
                                                                                                                  ^~~~~~~~~~~
/home/ashley/.cargo/registry/src/github.com-88ac128001ac3a9a/quasi-0.3.11/src/lib.rs:354:9: 354:38 note: in this expansion of panictry! (defined in /home/ashley/.cargo/registry/src/github.com-88ac128001ac3a9a/quasi-0.3.11/src/lib.rs)
/home/ashley/.cargo/registry/src/github.com-88ac128001ac3a9a/quasi-0.3.11/src/lib.rs:358:26: 358:36 error: cannot infer an appropriate lifetime for lifetime parameter `'a` due to conflicting requirements [E0495]
/home/ashley/.cargo/registry/src/github.com-88ac128001ac3a9a/quasi-0.3.11/src/lib.rs:358         panictry!(parser.parse_ty())
                                                                                                                  ^~~~~~~~~~
/home/ashley/.cargo/registry/src/github.com-88ac128001ac3a9a/quasi-0.3.11/src/lib.rs:358:9: 358:37 note: in this expansion of panictry! (defined in /home/ashley/.cargo/registry/src/github.com-88ac128001ac3a9a/quasi-0.3.11/src/lib.rs)
/home/ashley/.cargo/registry/src/github.com-88ac128001ac3a9a/quasi-0.3.11/src/lib.rs:357:58: 359:2 note: first, the lifetime cannot outlive the destruction scope surrounding block at 357:57...
/home/ashley/.cargo/registry/src/github.com-88ac128001ac3a9a/quasi-0.3.11/src/lib.rs:357 pub fn parse_ty_panic(parser: &mut Parser) -> P<ast::Ty> {
/home/ashley/.cargo/registry/src/github.com-88ac128001ac3a9a/quasi-0.3.11/src/lib.rs:358         panictry!(parser.parse_ty())
/home/ashley/.cargo/registry/src/github.com-88ac128001ac3a9a/quasi-0.3.11/src/lib.rs:359 }
/home/ashley/.cargo/registry/src/github.com-88ac128001ac3a9a/quasi-0.3.11/src/lib.rs:358:19: 358:36 note: ...so that references are valid when the destructor runs
/home/ashley/.cargo/registry/src/github.com-88ac128001ac3a9a/quasi-0.3.11/src/lib.rs:358         panictry!(parser.parse_ty())
                                                                                                           ^~~~~~~~~~~~~~~~~
/home/ashley/.cargo/registry/src/github.com-88ac128001ac3a9a/quasi-0.3.11/src/lib.rs:358:9: 358:37 note: in this expansion of panictry! (defined in /home/ashley/.cargo/registry/src/github.com-88ac128001ac3a9a/quasi-0.3.11/src/lib.rs)
/home/ashley/.cargo/registry/src/github.com-88ac128001ac3a9a/quasi-0.3.11/src/lib.rs:357:58: 359:2 note: but, the lifetime must be valid for the anonymous lifetime #2 defined on the block at 357:57...
/home/ashley/.cargo/registry/src/github.com-88ac128001ac3a9a/quasi-0.3.11/src/lib.rs:357 pub fn parse_ty_panic(parser: &mut Parser) -> P<ast::Ty> {
/home/ashley/.cargo/registry/src/github.com-88ac128001ac3a9a/quasi-0.3.11/src/lib.rs:358         panictry!(parser.parse_ty())
/home/ashley/.cargo/registry/src/github.com-88ac128001ac3a9a/quasi-0.3.11/src/lib.rs:359 }
/home/ashley/.cargo/registry/src/github.com-88ac128001ac3a9a/quasi-0.3.11/src/lib.rs:358:26: 358:36 note: ...so that types are compatible (expected `&mut syntax::parse::parser::Parser<'_>`, found `&mut syntax::parse::parser::Parser<'_>`)
/home/ashley/.cargo/registry/src/github.com-88ac128001ac3a9a/quasi-0.3.11/src/lib.rs:358         panictry!(parser.parse_ty())
                                                                                                                  ^~~~~~~~~~
/home/ashley/.cargo/registry/src/github.com-88ac128001ac3a9a/quasi-0.3.11/src/lib.rs:358:9: 358:37 note: in this expansion of panictry! (defined in /home/ashley/.cargo/registry/src/github.com-88ac128001ac3a9a/quasi-0.3.11/src/lib.rs)
/home/ashley/.cargo/registry/src/github.com-88ac128001ac3a9a/quasi-0.3.11/src/lib.rs:362:22: 362:34 error: cannot infer an appropriate lifetime for lifetime parameter `'a` due to conflicting requirements [E0495]
/home/ashley/.cargo/registry/src/github.com-88ac128001ac3a9a/quasi-0.3.11/src/lib.rs:362     panictry!(parser.parse_stmt())
                                                                                                              ^~~~~~~~~~~~
/home/ashley/.cargo/registry/src/github.com-88ac128001ac3a9a/quasi-0.3.11/src/lib.rs:362:5: 362:35 note: in this expansion of panictry! (defined in /home/ashley/.cargo/registry/src/github.com-88ac128001ac3a9a/quasi-0.3.11/src/lib.rs)
/home/ashley/.cargo/registry/src/github.com-88ac128001ac3a9a/quasi-0.3.11/src/lib.rs:362:15: 362:34 note: first, the lifetime cannot outlive the method call at 362:14...
/home/ashley/.cargo/registry/src/github.com-88ac128001ac3a9a/quasi-0.3.11/src/lib.rs:362     panictry!(parser.parse_stmt())
                                                                                                       ^~~~~~~~~~~~~~~~~~~
/home/ashley/.cargo/registry/src/github.com-88ac128001ac3a9a/quasi-0.3.11/src/lib.rs:362:5: 362:35 note: in this expansion of panictry! (defined in /home/ashley/.cargo/registry/src/github.com-88ac128001ac3a9a/quasi-0.3.11/src/lib.rs)
/home/ashley/.cargo/registry/src/github.com-88ac128001ac3a9a/quasi-0.3.11/src/lib.rs:362:15: 362:34 note: ...so type `fn(&mut syntax::parse::parser::Parser<'_>) -> core::result::Result<core::option::Option<syntax::ptr::P<syntax::codemap::Spanned<syntax::ast::Stmt_>>>, syntax::errors::DiagnosticBuilder<'_>>` of expression is valid during the expression
/home/ashley/.cargo/registry/src/github.com-88ac128001ac3a9a/quasi-0.3.11/src/lib.rs:362     panictry!(parser.parse_stmt())
                                                                                                       ^~~~~~~~~~~~~~~~~~~
/home/ashley/.cargo/registry/src/github.com-88ac128001ac3a9a/quasi-0.3.11/src/lib.rs:362:5: 362:35 note: in this expansion of panictry! (defined in /home/ashley/.cargo/registry/src/github.com-88ac128001ac3a9a/quasi-0.3.11/src/lib.rs)
/home/ashley/.cargo/registry/src/github.com-88ac128001ac3a9a/quasi-0.3.11/src/lib.rs:361:70: 363:2 note: but, the lifetime must be valid for the anonymous lifetime #2 defined on the block at 361:69...
/home/ashley/.cargo/registry/src/github.com-88ac128001ac3a9a/quasi-0.3.11/src/lib.rs:361 pub fn parse_stmt_panic(parser: &mut Parser) -> Option<P<ast::Stmt>> {
/home/ashley/.cargo/registry/src/github.com-88ac128001ac3a9a/quasi-0.3.11/src/lib.rs:362     panictry!(parser.parse_stmt())
/home/ashley/.cargo/registry/src/github.com-88ac128001ac3a9a/quasi-0.3.11/src/lib.rs:363 }
/home/ashley/.cargo/registry/src/github.com-88ac128001ac3a9a/quasi-0.3.11/src/lib.rs:362:22: 362:34 note: ...so that types are compatible (expected `&mut syntax::parse::parser::Parser<'_>`, found `&mut syntax::parse::parser::Parser<'_>`)
/home/ashley/.cargo/registry/src/github.com-88ac128001ac3a9a/quasi-0.3.11/src/lib.rs:362     panictry!(parser.parse_stmt())
                                                                                                              ^~~~~~~~~~~~
/home/ashley/.cargo/registry/src/github.com-88ac128001ac3a9a/quasi-0.3.11/src/lib.rs:362:5: 362:35 note: in this expansion of panictry! (defined in /home/ashley/.cargo/registry/src/github.com-88ac128001ac3a9a/quasi-0.3.11/src/lib.rs)
/home/ashley/.cargo/registry/src/github.com-88ac128001ac3a9a/quasi-0.3.11/src/lib.rs:366:26: 366:55 error: cannot infer an appropriate lifetime for lifetime parameter `'a` due to conflicting requirements [E0495]
/home/ashley/.cargo/registry/src/github.com-88ac128001ac3a9a/quasi-0.3.11/src/lib.rs:366         panictry!(parser.parse_attribute(permit_inner))
                                                                                                                  ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/ashley/.cargo/registry/src/github.com-88ac128001ac3a9a/quasi-0.3.11/src/lib.rs:366:9: 366:56 note: in this expansion of panictry! (defined in /home/ashley/.cargo/registry/src/github.com-88ac128001ac3a9a/quasi-0.3.11/src/lib.rs)
/home/ashley/.cargo/registry/src/github.com-88ac128001ac3a9a/quasi-0.3.11/src/lib.rs:366:19: 366:55 note: first, the lifetime cannot outlive the method call at 366:18...
/home/ashley/.cargo/registry/src/github.com-88ac128001ac3a9a/quasi-0.3.11/src/lib.rs:366         panictry!(parser.parse_attribute(permit_inner))
                                                                                                           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/ashley/.cargo/registry/src/github.com-88ac128001ac3a9a/quasi-0.3.11/src/lib.rs:366:9: 366:56 note: in this expansion of panictry! (defined in /home/ashley/.cargo/registry/src/github.com-88ac128001ac3a9a/quasi-0.3.11/src/lib.rs)
/home/ashley/.cargo/registry/src/github.com-88ac128001ac3a9a/quasi-0.3.11/src/lib.rs:366:19: 366:55 note: ...so type `fn(&mut syntax::parse::parser::Parser<'_>, bool) -> core::result::Result<syntax::codemap::Spanned<syntax::ast::Attribute_>, syntax::errors::DiagnosticBuilder<'_>>` of expression is valid during the expression
/home/ashley/.cargo/registry/src/github.com-88ac128001ac3a9a/quasi-0.3.11/src/lib.rs:366         panictry!(parser.parse_attribute(permit_inner))
                                                                                                           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/ashley/.cargo/registry/src/github.com-88ac128001ac3a9a/quasi-0.3.11/src/lib.rs:366:9: 366:56 note: in this expansion of panictry! (defined in /home/ashley/.cargo/registry/src/github.com-88ac128001ac3a9a/quasi-0.3.11/src/lib.rs)
/home/ashley/.cargo/registry/src/github.com-88ac128001ac3a9a/quasi-0.3.11/src/lib.rs:365:89: 367:2 note: but, the lifetime must be valid for the anonymous lifetime #2 defined on the block at 365:88...
/home/ashley/.cargo/registry/src/github.com-88ac128001ac3a9a/quasi-0.3.11/src/lib.rs:365 pub fn parse_attribute_panic(parser: &mut Parser, permit_inner: bool) -> ast::Attribute {
/home/ashley/.cargo/registry/src/github.com-88ac128001ac3a9a/quasi-0.3.11/src/lib.rs:366         panictry!(parser.parse_attribute(permit_inner))
/home/ashley/.cargo/registry/src/github.com-88ac128001ac3a9a/quasi-0.3.11/src/lib.rs:367 }
/home/ashley/.cargo/registry/src/github.com-88ac128001ac3a9a/quasi-0.3.11/src/lib.rs:366:26: 366:55 note: ...so that types are compatible (expected `&mut syntax::parse::parser::Parser<'_>`, found `&mut syntax::parse::parser::Parser<'_>`)
/home/ashley/.cargo/registry/src/github.com-88ac128001ac3a9a/quasi-0.3.11/src/lib.rs:366         panictry!(parser.parse_attribute(permit_inner))
                                                                                                                  ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/ashley/.cargo/registry/src/github.com-88ac128001ac3a9a/quasi-0.3.11/src/lib.rs:366:9: 366:56 note: in this expansion of panictry! (defined in /home/ashley/.cargo/registry/src/github.com-88ac128001ac3a9a/quasi-0.3.11/src/lib.rs)
error: aborting due to 8 previous errors
```

It looks like most of it's about lifetimes in the `panictry` macro.

**Comment by erickt** [comment]
Should be fixed in 0.3.12!

**Comment by KodrAus** [comment]
Awesome thanks! Would it be worth using something like https://nightli.es to force a build at least once a day so we know when changes in Rust Nightly break the build?
